Improve documentation and golint compliance of registry package

aaronlehmann opened the pull request:
- Add godoc documentation where it was missing
- Change identifier names that don't match Go style, such as INDEX_NAME
- Rename RegistryInfo to PingResult, which more accurately describes
  what this structure is for. It also has the benefit of making the name
  not stutter if used outside the package.

Updates #14756

Signed-off-by: Aaron Lehmann aaron.lehmann@docker.com

Commit:
Improve documentation and golint compliance of registry package

* Add godoc documentation where it was missing

* Change identifier names that don't match Go style, such as INDEX_NAME

* Rename RegistryInfo to PingResult, which more accurately describes
  what this structure is for. It also has the benefit of making the name
  not stutter if used outside the package.

Updates #14756

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

aaronlehmann commented:
@stevvooe @dmcgowan @tiborvass PTAL

I'm not 100% confident in the documentation I've added, and would appreciate a thorough review.

stevvooe commented on registry/session.go:
```diff
@@ -308,9 +315,10 @@ func (r *Session) GetRemoteImageLayer(imgID, registry string, imgSize int64) (io
 	return res.Body, nil
 }
 
+// GetRemoteTag retrieves a tag from the given repository
```
Let's make an attempt to describe the arguments. What does `askedTag` do?

dmcgowan replied:
`askedTag` refers to the tag as requested by the user. Usually the term asked is used when the tag may also be "empty", indicating a user did not ask for a tag but one was defaulted or multi tags were used. In this case the variable should likely just be named `tag`.

stevvooe commented on registry/session.go:
```diff
@@ -331,18 +339,19 @@ func (r *Session) GetRemoteTag(registries []string, repository string, askedTag
 	return "", fmt.Errorf("Could not reach any registry endpoint")
 }
 
+// GetRemoteTags retrieves all tags from the given repository
```
What are the contents of the returned map?

dmcgowan replied:
Excellent question, the `result` variable doesn't appear to line up with what is documented https://docs.docker.com/reference/api/hub_registry_spec/#tags-registry

stevvooe commented on registry/types.go:
```diff
@@ -1,26 +1,43 @@
+	Tokens []string
 }
 
+// ImgData describes an image
```
Do we know what purpose this serves? hint: where is this type used?

dmcgowan replied:
This type is used by the v1 code to store the list of images which will be pushed or pulled. My understanding is the "Checksum" in this struct has no significant use.

stevvooe commented on registry/types.go:
```diff
@@ -1,26 +1,43 @@
 
+// ImgData describes an image
 type ImgData struct {
 	ID              string `json:"id"`
```
Document these fields?

aaronlehmann replied:
I documented ID, but I couldn't quite figure out the other fields. The checksum fields are particularly confusing. Do you have any insight into these?

stevvooe commented on registry/types.go:
```diff
@@ -28,11 +45,21 @@ type ImgData struct {
+// standalone registry.
+type PingResult struct {
+	// Version is the registry version supplied in the
+	// X-Docker-Registry-Version header
```
The header name may be decoupled from the version. For example, in version 2, we use `Docker-Registry-Api-Version` or something like that.

stevvooe commented:
@aaronlehmann Mostly nits. Excellent first pass.

tiborvass commented:
@aaronlehmann sorry need a rebase :S

aaronlehmann commented:
Rebased and pushed new changes based on review comments. I made the review fixes a separate commit for ease of review. I'll squash the commits before this gets merged.

vdemeester commented:
@aaronlehmann I think if you rebase (or have rebased) you could also add the packages covered in ``./hack/make/validate-lint`` :blush: .

aaronlehmann commented:
Rebased and pushed again. Added the package to `hack/make/validate-lint`, and fixed two more lint warnings that crept in during the last day.

Let me know when you are ready for me to squash these commits for merge, or if there are additional things to improve.

stevvooe commented:
LGTM

tiborvass commented:
@aaronlehmann awfully sorry, needs another rebase, will make sure it gets merged asap

aaronlehmann commented:
Rebased and squashed.

tiborvass commented on registry/registry.go:
```diff
@@ -98,6 +93,8 @@ func trustedLocation(req *http.Request) bool {
 
+// AddRequiredHeadersToRedirectedRequests adds the necessary redirection headers
+// for redirected requests
 func AddRequiredHeadersToRedirectedRequests(req *http.Request, via []*http.Request) error {
```
@aaronlehmann doesn't need to be exported at all

tiborvass commented:
LGTM

tiborvass commented:
Only waiting for compilation on CI coz its only comments, and the var renamings would have surfaced under compiling errors.

tiborvass commented:
@aaronlehmann forgot to change AddRequire... in registry_test

tiborvass commented:
Tested locally, merging
